railties/test/commands/server_test.rb

```diff
@@ -118,4 +118,18 @@ def test_default_options
       assert_equal old_default_options, server.default_options
     end
   end
+
+  def test_restart_command_contains_customized_options
+    original_args = ARGV
+    args = ["-p", "4567"]
+    ARGV.replace args
+
+    options = Rails::Server::Options.new.parse! args
+    server = Rails::Server.new options
+    expected = "bin/rails server -p 4567"
+
+    assert_equal expected, server.default_options[:restart_cmd]
+  ensure
+    ARGV.replace original_args
```

Comment:
The other option is to use `ARGV[0..-1] = args` instead of `ARGV.replace`. 

@rafaelfranca What do you think about `ARGV[0..-1] = args`?